### PR TITLE
Feature/issue 60

### PR DESCRIPTION
**Pull request by merry-degaga**
Hi Karan, I broke up this issue into subtasks to make sure each gets your approval before moving onto the next subtask. 

- [x]  Add a button in the header which says 'Scanners' 
- [x]  Button is clicked and opens up a menu bar with two options (one for DAST and one for SAST)
- [ ]  DAST menu item will provide info on how DAST scanners can use VulnerableApp-facade
- [ ]  SAST menu item will provide info on how SAST scanners can use VulnerableApp-facade
- [ ]  Add merge logic in LUA

This pull request contains the first two items from above. Specifically what has been done is:
- Added tests in Header test file
- Edited Header component to include Scanners menu
- Added css to App.css so child menu items of Scanner menu are the same width as the parent 
- Updated the Header snapshot

If these changes look good so far, I will move onto the next subtask.

Some questions:
- Would it be more preferable to create a styles folder and Header.css instead instead of using App.css?
- For DAST info, how is the flow of the application? Is it when a user clicks on the 'DAST' menu item, it directs them to a page within the application (so I would need to create a new component) or is it sufficient to just link to 'https://sasanlabs.github.io/VulnerableApp/HOW-TO-USE.html#how-sast-or-dast-can-use-the-project' ?
- Is it okay to leave the -u in the package.json for the 'test' script? Helped with the failed pipelines.

Thank you.

### CODE REVIEW

**Comment by preetkaran20** [comment]
> 
> * Would it be more preferable to create a styles folder and Header.css instead instead of using App.css?
yes, Great thought

> * For DAST info, how is the flow of the application? Is it when a user clicks on the 'DAST' menu item, it directs them to a page within the application (so I would need to create a new component) or is it sufficient to just link to 'https://sasanlabs.github.io/VulnerableApp/HOW-TO-USE.html#how-sast-or-dast-can-use-the-project' ?

I think it is better to have all the data in Facade app, we will fetch the details by calling endpoints on each application and merge it in the Nginx lua script, then we can add the content explaining about how to use that data and few blog/video links after that.

> * Is it okay to leave the -u in the package.json for the 'test' script? Helped with the failed pipelines.
I am not sure about it. can you please explain what does `-u` do ?

File: facade-app/src/Components/Header.tsx [hunk 1]
```diff
@@ -93,6 +93,10 @@ export default class Header extends React.Component<Props, {}> {
                   </RSuiteDropDown.Item>
                 </a>
               </RSuiteDropDown>
+              <RSuiteDropDown title="Scanners">
+                <RSuiteDropDown.Item>DAST</RSuiteDropDown.Item>
```
Comment (preetkaran20) [inline facade-app/src/Components/Header.tsx:97]: can we add a tooltip such that on Hover of DAST it shows the fullform of DAST and similarly for SAST as well?

File: facade-app/src/Components/Header.tsx [hunk 2]
```diff
@@ -93,6 +93,10 @@ export default class Header extends React.Component<Props, {}> {
                   </RSuiteDropDown.Item>
                 </a>
               </RSuiteDropDown>
+              <RSuiteDropDown title="Scanners">
```
Comment (preetkaran20) [inline facade-app/src/Components/Header.tsx:96]: can we move Scanners after aboutus header and let github and projects by SasanLabs at the end?

**Review by preetkaran20 — APPROVED** [review]
LGTM, just 2 small change
1. Tooltip for SAST and DAST with fullform
2. Scanners nav item just after Aboutus

**Comment by merry-degaga** [comment]
- Added tests for DAST and SAST tooltips, updated Header component
- Moved Scanners dropdown to after AboutUs
- Created style/Header.css

**Comment by merry-degaga** [comment]
> > * Would it be more preferable to create a styles folder and Header.css instead instead of using App.css?
> 
> Yes, Great thought
> 
> > * For DAST info, how is the flow of the application? Is it when a user clicks on the 'DAST' menu item, it directs them to a page within the application (so I would need to create a new component) or is it sufficient to just link to 'https://sasanlabs.github.io/VulnerableApp/HOW-TO-USE.html#how-sast-or-dast-can-use-the-project' ?
> 
> I think it is better to have all the data in Facade app, we will fetch the details by calling endpoints on each application and merge it in the Nginx lua script, then we can add the content explaining about how to use that data and few blog/video links after that.
> 
> > * Is it okay to leave the -u in the package.json for the 'test' script? Helped with the failed pipelines.
> 
> I am not sure about it. can you please explain what does `-u` do ?

Yes, the -u is shorthand for <a href="https://jestjs.io/docs/snapshot-testing">updateSnapshot</a>. 
Initally when updating the Header component, the snapshot test in the Header test file was failing, so used that to help pass that test. 
After I submitted a merge request, the compile action failed due to the snapshot not matching for 'src/test/App.test.tsx'. I did not see this file locally but it appears during the compile action and adding -u in package.json resolved the App 's snapshot test.

**Comment by merry-degaga** [comment]
> > * Added tests for DAST and SAST tooltips, updated Header component
> > * Moved Scanners dropdown to after AboutUs
> > * Created style/Header.css
> 
> LGTM, shall we merge this PR and you will raise other PR's with the pending tasks or do you want to continue working on remaining tasks under this PR only?

Okay, yes this can be mered and will work on the remaining tasks and make separate PR's for those.

**Pull request merged**